### PR TITLE
chore: update cypress to 12.8.1

### PR DESCRIPTION
**Pull request by AtofStryker**
bumps cypress minor version to 12.8.1 and bumps circle ci docker images to newer browsers

### CODE REVIEW

**Comment by AtofStryker** [comment]
Looks like there is a regression in the notification spec for firefox, but it looks like it has been here before this commit

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -40,6 +40,9 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
```
Comment (AtofStryker) [inline .github/workflows/main.yml:44]: the runner in this job installs node 18, which I believe has some compatibility issues with openSSL inside this windows job. I downgraded it to node 16 to get the build to pass ([stackoverflow for reference](https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported))

Comment (mschile) [reply .github/workflows/main.yml:44]: Have we tried running on [windows server 2022](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)? Would also suggest adding a comment here so we know why node 16 was installed when we try to upgrade.

Comment (AtofStryker) [reply .github/workflows/main.yml:44]: I have not tried windows server 2022. I think the node version is likely similar so I can pin it and add a comment and try the upgrade to 2022 as well

**Review by mschile — APPROVED** [review]

**Pull request merged**